lib/utils.js

```diff
@@ -96,6 +96,9 @@ function patchEventTargetMethods(obj) {
   // This is required for the addEventListener hook on the root zone.
   obj[keys.common.addEventListener] = obj.addEventListener;
   obj.addEventListener = function (eventName, handler, useCapturing) {
+    if (!handler || (typeof(handler) !== 'function' && typeof(handler.handleEvent) !== 'function')) {
+      return;
```

Comment:
I'm am not sure how this PR relates to #167, could you please expand ?

I believe that `!handler` is the only condition that could make zone.js code fail. Any other value should not make zone.js fail and then there is no reason why we should not call the underlying `xEventListener` method - zone.js should not try to fix its users errors.

The less code we have the less bugs we have. For ex I think your code might fail on IE because of the `FunctionWrapper` thing (when dev tools are open, the handler could be a `FunctionWrapper` IIRC, see a few lines below your updates).

That being said, there might be some obvious thing I fail to get here ?

wrt to your tests, what would fail ?

Also please update the tests to reflect the original issue:

> It may throw some errors when there were no handler to remove.

```js
expect(function() {
  button.addEventListener('click', eventListenerA);
}).not.toThrowError();
```